### PR TITLE
Faker data generation

### PR DESCRIPTION
**Pull request by Aboorva1**
Used faker gem to generate data 
Made some design changes
Added validation in options model so that more than 4 options cannot be added for a question 
A question can have three options as false and only one option as true

### CODE REVIEW

File: app/models/option.rb [hunk 1]
```diff
@@ -1,6 +1,39 @@
...
+
+  def options_count   
+    options = Option.where(question_id: question_id)
+    if options.count >= 4
+      errors.add(:base, "Question should have exactly 4 options")
+    end
+  end
 
+  def correct_answer_count
+
```
Comment (iMagesh) [inline app/models/option.rb]: why is there a line space here?

File: app/models/question.rb [hunk 2]
```diff
@@ -1,8 +1,10 @@
 class Question < ApplicationRecord
 	belongs_to :quiz
```
Comment (iMagesh) [inline app/models/question.rb:2]: is there an alignment problem here?

File: app/models/quiz.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 class Quiz < ApplicationRecord
     belongs_to :category
     belongs_to :user, optional: true
-    validates :title, presence: true, length: { in: 8..20 }
+    validates :title, presence: true
```
Comment (iMagesh) [inline app/models/quiz.rb]: why did you remove the length validation for title?

File: app/models/user.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 class User < ApplicationRecord
-  has_many :quizzes
-  has_many :user_answers
-  has_many :user_quizzes
+  has_many :quizzes, :dependent => :destroy
+  has_many :user_answers, :dependent => :destroy
+  has_many :user_quizzes, :dependent => :destroy
```
Comment (iMagesh) [inline app/models/user.rb:4]: can you move all of this to one line?

File: app/views/leader_board/index.html.erb [hunk 5]
```diff
@@ -15,21 +15,22 @@
...
-        </p>
+        <div class="pl-lg-5 pl-md-3 pl-1 name"><%= user_quiz.user.name %></div>
       </td> 
       <td><%= user_quiz.quiz.title %></td> 
       <td><%= user_quiz.score %></td>   
       <td><%= user_quiz.created_at.strftime("%d %b %Y") %></td>      
     </tr>
+    <tr id="spacing-row">
+                    <td></td>
+                </tr>
```
Comment (iMagesh) [inline app/views/leader_board/index.html.erb]: is this aligned correctly?

**Review by iMagesh — CHANGES_REQUESTED** [review]
Please respond to all the comments, let's fix it all right now and merge.

File: app/models/option.rb (hunk 1 above)
Comment (Aboorva1) [reply app/models/option.rb]: Removed the line space

File: app/models/question.rb (hunk 2 above)
Comment (Aboorva1) [reply app/models/question.rb:2]: It is aligned properly in code

File: app/models/user.rb (hunk 4 above)
Comment (Aboorva1) [reply app/models/user.rb:4]: Tried it but it is giving error

File: app/views/leader_board/index.html.erb (hunk 5 above)
Comment (Aboorva1) [reply app/views/leader_board/index.html.erb]: Fixed the alignment

File: app/models/user.rb (hunk 4 above)
Comment (iMagesh) [reply app/models/user.rb:4]: maybe it is not possible

**Pull request merged**